### ISSUE TITLE
Snap Shot

### ISSUE THREAD

**Issue by Theoden91** [xws-bench/bench#81]
There's a few problems with 'Snap Shot' EPT. First of all it activates each time an enemy ship ends a movement in Firing Arc in all ranges (not only in R1). Second of all if you activate it in such situation, whole game just freezes - no further windows popup and show. You just have to back to previous turn and hope you don't make that mistake again.

**Comment by echoesandechoes** [comment]
Can confirm.

**Comment by Sleet01** [comment]
There are a number of problems with Snap Shot right now.
1) It uses the ship's default weapon, rather than adding itself as a new secondary weapon.
2) It does not check range.
3) It attaches a new function to *every* ship in the game, not just all enemy ships.  This function is used to "wake up" the Snap Shot carrier every time another ship ends its movement, and prompt to see if the Snap Shot ship wants to attack.
4) It doesn't differentiate between "phases" and "rounds", so if you fire Snap Shot, that ship can't fire again for a whole round.

I believe this functionality would be better implemented using a custom JQuery event handler on the Snap Shot ship, and attaching a post-movement Event thrower on every enemy ship.  But I don't have a lot of time to try this out right now.

**Comment by echoesandechoes** [comment]
Good catch on all these.

Correction to your 4th point though: Snap Shot can be activated once per movement phase. The Snap Shot ship may then still fire during combat phase. ([See the X-Wing wiki for card text.](http://xwing-miniatures.wikia.com/wiki/Snap_Shot))

**Comment by Sleet01** [comment]
Right, what I mean is, the current implementation will let you Snap Shot as many times as you want during the Movement Phase,  but if you take even one Snap Shot attack during the movement phase that ship is not allowed to fire its default weapon for the rest of the turn.  You *should* be able to fire (iirc) a Snap Shot during the Movement Phase,  a regular shot during the Attack phase, and a Snap Shot during any of the later phases  (provided an enemy can perform a maneuver somehow).  Currently none of that is working correctly
